contrib/debian/rules

```diff
@@ -22,13 +22,25 @@ ifeq (yes,$(shell dpkg-vendor --derives-from Ubuntu && echo yes))
 	SB_STYLE := ubuntu
 	tar_name := fwupd_$(deb_version)_$(DEB_HOST_ARCH).tar.gz
 	CONFARGS += -Dplugin_flashrom=false
+	CONFARGS += -Defi_sbat_distro_id=ubuntu
+	CONFARGS += -Defi_sbat_distro_generation=1
+	CONFARGS += -Defi_sbat_distro_summary=Ubuntu
+	CONFARGS += -Defi_sbat_distro_pkgname=fwupd
+	CONFARGS += -Defi_sbat_distro_version=$(deb_version)
+	CONFARGS += -Defi_sbat_distro_url="https://packages.ubuntu.com/search?keywords=fwupd"
 else
 	TMPLDIR  := debian/fwupd-$(DEB_HOST_ARCH)-signed-template/usr/share/code-signing/fwupd-$(DEB_HOST_ARCH)-signed-template
 	ifneq ($(DEB_HOST_ARCH_CPU),ia64)
 		CONFARGS += -Dplugin_flashrom=true
 	else
 		CONFARGS += -Dplugin_flashrom=false
 	endif
+	CONFARGS += -Defi_sbat_distro_id=debian
```

Comment:
I wonder if it's worth just parsing /etc/os-release directly in case a derivative decides to rebuild fwupd they don't end up with claiming to be debian.

Reply:
What I want to happen is the spec changes to an AppStream ID, so rather than `fwupd.fedora` we'd have the java-style `org.fedoraproject.fwupd` if I can convince @vathpela that it's a good idea. And that, we can't get from the os-release file without some serious futzing.

Reply:
My concern here is that we need the `SBAT` UEFI variable to be /small/, and I already think the strings we've currently got are pushing it.  If we wanted to use something more verbose in the `.sbat` section, we probably need a mapping table somewhere.  Which means we want some canonical source of the aliases that wind up in `SBAT` and all that goes with that.

Reply:
If we want it to be small - why bother including a URL in the SBAT field?  That is certainly the biggest part right now, and people should be able to encode that metadata elsewhere from the actual binary itself.

Reply:
> we need the SBAT UEFI variable to be /small

The URLs are the biggest strings by far.

Reply:
Those aren't in the variable at all.

Reply:
Ohh, I see, sorry. My brain assumed the entire .sbat section would be in the variable, but that's not the case at all.

Reply:
Mine made the same conclusion until I re-read the comment above.

Reply:
> My concern here is that we need the SBAT UEFI variable to be /small/

Could we define three letter shorthands for the different distros? e.g. RHT = Red Hat, FED = Fedora etc? That obviously needs a "registry" of sorts, but I don't think that's always a bad thing.

Reply:
I think something that could be done is to mention in the spec that the `component_name` should be the existing vendor EFI directory name and the component name. 

i.e `fedora.grub2`, `redhat.fwupd` and so on, which is what @vathpela already has in his proposal. This seems to be a middle ground between a full reverse DNS identification string and just an arbitrary string for the distro. The EFI vendor directory should be exclusive anyways or otherwise two distros would write their EFI binaries in the same directory in the ESP.

Reply:
> component_name should be the existing vendor EFI directory name

If we have that in the spec, we can just reuse the existing `-Defi_os_dir` define.

Reply:
> If we have that in the spec, we can just reuse the existing -Defi_os_dir define.

I've respun the patch to use that.